paddle/fluid/framework/details/op_handle_base.cc

```diff
@@ -122,35 +122,17 @@ void OpHandleBase::RunAndRecordEvent(const std::function<void()> &callback) {
 #ifdef PADDLE_WITH_CUDA
   if (!events_.empty()) {  // Use event
     std::function<void()> method = callback;
-
+    // NOTE(zcd): device context must be ordered here because RecordEvent
+    // will use a mutex to ensure the safe of multi-threads.
+    std::map<platform::DeviceContext *, platform::Place> ordered_ctxes;
     for (auto &p : dev_ctxes_) {
```

Comment:
perhaps we should make ```dev_ctxes_``` sorted in the future. ```RunAndRecordEvent``` has been overwritten in many op handles.

Reply:
Thanks, I will fix it in next PR.